gogol/src/Network/Google/Auth.hs

```diff
@@ -110,6 +128,33 @@ initStore :: (MonadIO m, MonadCatch m, AllowScopes s)
           -> m (Store s)
 initStore c l m = exchange c l m >>= fmap Store . liftIO . newMVar
 
+-- | Refresh an 'OAuthToken' if it was expired.
+-- Otherwise, keep original one.
+refreshStore :: (MonadIO m, MonadCatch m, AllowScopes s)
+             => Store s
+             -> Logger
+             -> Manager
+             -> m (Store s)
+refreshStore ss@(Store s) l m = do
+    x  <- retrieveAuthFromStore ss
+    mx <- validate x
+    if mx
+        then pure ss
+        else liftIO . modifyMVar s $ \y -> do
```

Comment:
Hmmm.. after re-tracking the code, I think this function might be unnecessary. It looks like the token would be refreshed if it expired while using `send`, isn't it? so, I removed it for now.